core/include/scipp/core/variable_keyword_arg_constructor.h

```diff
@@ -128,18 +128,26 @@ template <class VarT, class... Ts> class ConstructorArgumentsMatcher {
         "Shape{1, 2}, Dims{Dim::X, Dim::Y}, Values({3, 4}))");
   }
 
-  template <class ElemT, class... NonDataTypes>
-  static VarT construct(Ts &&... ts) {
+  template <class... NonDataTypes> static auto extractArguments(Ts &&... ts) {
     auto tp = std::make_tuple(std::forward<Ts>(ts)...);
-    auto valArgs = std::move(extractTagged<ValuesTag, Ts...>(tp));
-    auto varArgs = std::move(extractTagged<VariancesTag, Ts...>(tp));
+    return std::make_tuple(std::move(extractTagged<ValuesTag, Ts...>(tp)),
+                           std::move(extractTagged<VariancesTag, Ts...>(tp)),
+                           std::tuple<NonDataTypes...>(std::move(
```

Comment:
I don't understand, why do you need to move the return value?

Changes look good otherwise 👍

Reply:
`extraxtArgs` retuns a lvalue reference to something in `tp` if there is something, so `move` is to avoid copy.

Reply:
Why not move in `extractArgs`, that would be clearer?